Spatial.gd

```diff
@@ -929,7 +935,14 @@ func _on_NewIcon_pressed():
 	position.set_x(self.player_position.x)
 	position.set_y(self.player_position.y)
 	position.set_z(-self.player_position.z)
-	gen_new_icon(self.next_texture_trail, waypoint_icon, self.currently_active_category)
+	var texture_id: int = get_texture_index(self.next_texture_path)
+	if texture_id != -1:
+		waypoint_icon.set_texture_id(texture_id)
+	else:
+		self.waypoint_data.get_textures().add_textures().set_filepath(self.next_texture_path)
+		texture_id = self.waypoint_data.get_textures().add_textures().size() - 1
+		waypoint_icon.set_texture_id(texture_id)
```

Comment:
This wont be necessary here if we already have a `self.next_texture_index`

Reply:
Agreed